File: src/main/java/tn/esprit/entities/Abonnement.java
package tn.esprit.entities;

public class Abonnement {
    private int id_Abonnement, id_Utilisateur, id_Pack;
    private String date_Souscription, date_Expiration, statut;

    // Constructeur avec paramètres
    public Abonnement(int id_Abonnement, int id_Utilisateur, int id_Pack, String date_Souscription, String date_Expiration, String statut) {
        this.id_Abonnement = id_Abonnement;
        this.id_Utilisateur = id_Utilisateur;
        this.id_Pack = id_Pack;
        this.date_Souscription = date_Souscription;
        this.date_Expiration = date_Expiration;
        this.statut = statut;
    }

    // Constructeur sans paramètres
    public Abonnement() {
    }

    // Getters et Setters
    public int getId_Abonnement() {
        return id_Abonnement;
    }

    public void setId_Abonnement(int id_Abonnement) {
        this.id_Abonnement = id_Abonnement;
    }

    public int getId_Utilisateur() {
        return id_Utilisateur;
    }

    public void setId_Utilisateur(int id_Utilisateur) {
        this.id_Utilisateur = id_Utilisateur;
    }

    public int getId_Pack() {
        return id_Pack;
    }

    public void setId_Pack(int id_Pack) {
        this.id_Pack = id_Pack;
    }

    public String getDate_Souscription() {
        return date_Souscription;
    }

    public void setDate_Souscription(String date_Souscription) {
        this.date_Souscription = date_Souscription;
    }

    public String getDate_Expiration() {
        return date_Expiration;
    }

    public void setDate_Expiration(String date_Expiration) {
        this.date_Expiration = date_Expiration;
    }

    public String getStatut() {
        return statut;
    }

    public void setStatut(String statut) {
        this.statut = statut;
    }

    @Override
    public String toString() {
        return "Abonnement{" +
                "id_Abonnement=" + id_Abonnement +
                ", id_Utilisateur=" + id_Utilisateur +
                ", id_Pack=" + id_Pack +
                ", date_Souscription='" + date_Souscription + '\'' +
                ", date_Expiration='" + date_Expiration + '\'' +
                ", statut='" + statut + '\'' +
                '}';
    }
}

File: ProjetWebTrekSwap/src/Entity/Abonnement.php
<?php

namespace App\Entity;

use App\Repository\AbonnementRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: AbonnementRepository::class)]
class Abonnement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id_abonnement = null;

    #[ORM\Column(nullable: true)]
    private ?int $id_utilisateur = 0;

    #[ORM\Column]
    private ?int $id_Pack = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\GreaterThanOrEqual(value: "today", message: "La date de souscription ne peut pas être dans le passé.")]
    private ?\DateTimeInterface $date_Souscription = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\GreaterThan(propertyPath: "date_Souscription", message: "La date d'expiration ne peut pas être avant la date de souscription.")]
    private ?\DateTimeInterface $date_Expiration = null;

    #[ORM\Column(type: "string", length: 20)]
    private ?string $statut = null;

    public function getIdAbonnement(): ?int
    {
        return $this->id_abonnement;
    }

    public function setIdAbonnement(int $id_abonnement): static
    {
        $this->id_abonnement = $id_abonnement;

        return $this;
    }

    public function getIdUtilisateur(): ?int
    {
        return $this->id_utilisateur;
    }

    public function setIdUtilisateur(?int $id_utilisateur): static
    {
        $this->id_utilisateur = $id_utilisateur;

        return $this;
    }

    public function getIdPack(): ?int
    {
        return $this->id_Pack;
    }

    public function setIdPack(?int $id_Pack): static
    {
        $this->id_Pack = $id_Pack;

        return $this;
    }

    public function getDateSouscription(): ?\DateTimeInterface
    {
        return $this->date_Souscription;
    }

    public function setDateSouscription(?\DateTimeInterface $date_Souscription): static
    {
        $this->date_Souscription = $date_Souscription;

        return $this;
    }

    public function getDateExpiration(): ?\DateTimeInterface
    {
        return $this->date_Expiration;
    }

    public function setDateExpiration(?\DateTimeInterface $date_Expiration): static
    {
        $this->date_Expiration = $date_Expiration;

        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(?string $statut): static
    {
        $this->statut = $statut;

        return $this;
    }
}
